mrmustard/lab_dev/circuit_components.py

```diff
@@ -368,45 +387,77 @@ def on(self, modes: Sequence[int]) -> CircuitComponent:
             if subset and len(subset) != len(modes):
                 raise ValueError(f"Expected ``{len(modes)}`` modes, found ``{len(subset)}``.")
         ret = self._light_copy()
-        modes = set(modes)
         ret._wires = Wires(
-            modes_out_bra=modes if ob else set(),
-            modes_in_bra=modes if ib else set(),
-            modes_out_ket=modes if ok else set(),
-            modes_in_ket=modes if ik else set(),
+            modes_out_bra=set(modes) if ob else set(),
+            modes_in_bra=set(modes) if ib else set(),
+            modes_out_ket=set(modes) if ok else set(),
+            modes_in_ket=set(modes) if ik else set(),
         )
 
         return ret
 
-    def to_fock(self, shape: Optional[Union[int, Iterable[int]]] = None) -> CircuitComponent:
+    def fock(self, shape: Optional[int | Sequence[int]] = None) -> CircuitComponent:
```

Comment:
Ok, I remember thinking about this, and I concluded that I wanted to avoid padding with zeros and I'd rather chop the representations that will contract with this one instead. But you are right that it can be unexpected to put some numbers in and then get a different shape out. What if we raise an exception and return the smaller array? Then the user can make it bigger if they want to, but I don't want MM to be able to do the 'wrong thing' (to avoid uncaught regressions).

Reply:
What if the argument was called `max_shape` to hint that this will return the most compact thing and users only have the option to make it even more compact?

Reply:
that's another option. I'll think about it.

Reply:
I went for padding and warning